modin/experimental/core/execution/native/implementations/hdk_on_native/__init__.py

```diff
@@ -12,3 +12,102 @@
 # governing permissions and limitations under the License.
 
 """Experimental Modin's functionality related to Native execution engine and optimized for HDK storage format."""
+
+import re
+import sys
+import modin.pandas as pd
+from modin.pandas.utils import from_arrow
+from modin.experimental.core.storage_formats.hdk import DFAlgQueryCompiler
+from modin.experimental.core.execution.native.implementations.hdk_on_native.hdk_worker import (
+    HdkWorker,
+)
+
+
+def hdk_query(query: str) -> pd.DataFrame:
```

Comment:
what about putting the functionality to so-called "public" namespace ([`/modin/experimental/sql`](https://github.com/modin-project/modin/tree/master/modin/experimental/sql))?

The `/sql` module already contains the `query` function which as I understand provides very limited functionality in comparison with what we can achieve using HDK. May we define our own `query` function in the `/sql` module and dispatch between the `dfsql.query` and `hdk_query` depending on the execution selected?

Reply:
@ienkovich You are right, this is even a better approach.
In case if we get rid of the column names mangling, we will not need to import the tables and, thus, SQL parsing would still be required. The current hdk_query format is much easier for parsing. I would prefer to make it default, but keep support for the dfsql format, for backward compatibility.

Reply:
We can try to add `WITH` statements to SQL to make table aliases and thus avoid both SQL parsing and double-import problem. If the `WITH` approach doesn't work well, we can also introduce HDK API to create table aliases to avoid additional data import. We control this library, so we can introduce reasonable interfaces if we need them.

I know the current parsing looks simple, but it doesn't cover all cases. The first query with a string literal having `@` symbol would break it and require parser improvement. Then we would meet a case where two frames reference the same partition and that would break the resulting SQL again and require much more sophisticated parsing/substitution one more time. `dfsql` syntax just requires proper aliases to be created and it sounds like a much simpler task.

Reply:
> We can try to add `WITH` statements to SQL to make table aliases and thus avoid both SQL parsing and double-import problem.

Good idea! I'll check if it works.

> The first query with a string literal having `@` symbol would break it and require parser improvement

I thought about it. The current implementation throws exception in case if the referenced variable is not found in the context. The easiest solution would be to do not throw, but leave the string as is, but it may not work, if the variable exists, but substitution is not required for this string. In this case, some escape character should be introduced, that would definitely complicate the parsing.

> Then we would meet a case where two frames reference the same partition and that would break the resulting SQL

Could you please provide an example so I could check?

Reply:
@ienkovich WITH statements work! I've reworked the implementation.

Reply:
> Could you please provide an example so I could check?

I thought about something like this:

```
SELECT df1.x, df2.z FROM df1 LEFT JOIN df2 ON df1.x = df2.y;
```

Here `df1` and `df2` might reference the same dataframe and therefore the same partition. If we don't re-import this partition twice and use the existing HDK table, then `df1` and `df2` would be replaced with the same table name and SQL query would become invalid.

Reply:
@ienkovich Thanks. I checked this use case and it works with the new implementation, based on the WITH statement.